drivers/potentiometer/ad5293/ad5293.c

```diff
@@ -0,0 +1,531 @@
+	if (ret < 0)
+		return ret;
+
+	no_os_mdelay(2); // t12: max 1.5ms reset time
+
+	ad5293_reset_chip_info(dev);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Reset ad5293 through software reset command.
+ *
+ * @param dev - The device structure.
+ *
+ * @return ret - return code.
+ *         Example: -EINVAL - No valid pointer input.
+ *                  -ENOSYS - No valid operation.
+ *                  0 - No errors encountered.
+ *******************************************************************************/
+int32_t ad5293_soft_reset(struct ad5293_dev* dev)
+{
+	int32_t ret = 0;
+	uint8_t i;
+	uint16_t sz = 2 * dev->chip_num;
+	uint8_t buf[sz];
+
+	for (i = 0; i < dev->chip_num; i++) {
+		buf[2 * i] = (uint8_t)(RESET_FRAME >> 8);
+		buf[2 * i + 1] = (uint8_t)(RESET_FRAME);
+	}
+
+	ret = no_os_spi_write_and_read(dev->spi_desc, & buf[0], sz);
+	if (ret < 0)
+		return ret;
+
+	no_os_mdelay(2); // t12: max 1.5ms reset time
+
+	ad5293_reset_chip_info(dev);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Place SDO pin in high impedence mode.
+ *
+ * Minimize power dissipation when pin is used.
+ *
+ * @param dev - The device structure.
+ *
+ * @return ret - return code.
+ *         Example: -EINVAL - No valid pointer input.
+ *                  -ENOSYS - No valid operation.
+ *                  0 - No errors encountered.
+ *******************************************************************************/
+int32_t ad5293_sdo_setfloat(struct ad5293_dev* dev)
+{
+	int32_t ret = 0;
+	uint8_t i;
+	uint16_t sz = 2 * dev->chip_num;
+	uint8_t buf[sz];
+
+	for (i = 0; i < dev->chip_num; i++) {
+		buf[2 * i] = (uint8_t)(CMD_SDO_Z >> 8);
+		buf[2 * i + 1] = (uint8_t)CMD_SDO_Z;
+	}
+
+	ret = no_os_spi_write_and_read(dev->spi_desc, & buf[0], sz);
+	if (ret < 0)
+		return ret;
+
+	memset(& buf, 0, sz);
+	ret = no_os_spi_write_and_read(dev->spi_desc, & buf[0], sz);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Place ad5293 in shutdown mode or recovery.
+ *
+ * In shut down mode, terminal A open, Wiper and terminal B connected.
+ *
+ * @param dev - The device structure.
+ * @param sd - shutdown mode selection.
+ *
+ * @return ret - return code.
+ *         Example: -EINVAL - No valid pointer input.
+ *                  -ENOSYS - No valid operation.
+ *                  0 - No errors encountered.
+ *******************************************************************************/
+int32_t ad5293_shutdown(struct ad5293_dev* dev, enum shutdown_t sd)
+{
+	int32_t ret = 0;
+	uint8_t i;
+	uint16_t sz = 2 * dev->chip_num;
+	uint8_t buf[sz];
+
+	for (i = 0; i < dev->chip_num; i++) {
+		buf[2 * i] = (uint8_t)(SHUTDOWN_FRAME(sd) >> 8);
+		buf[2 * i + 1] = (uint8_t)SHUTDOWN_FRAME(sd);
+	}
+
+	ret = no_os_spi_write_and_read(dev->spi_desc, & buf[0], sz);
```

Comment:
same here, return on the same line, no need for `ret` variable, please apply to all if there are other instances

Reply:
will check all function